solr/core/src/java/org/apache/solr/search/MultiThreadedSearcher.java

```diff
@@ -255,7 +265,11 @@ public Object reduce(Collection collectors) throws IOException {
       for (Iterator var4 = collectors.iterator();
           var4.hasNext();
           maxScore = Math.max(maxScore, collector.getMaxScore())) {
-        collector = (MaxScoreCollector) var4.next();
+        Collector next = (Collector) var4.next();
```

Comment:
could use a comment like "assume the next collector, except EarlyTerminating, is the MaxScoreCollector".

While you are changing this code, please rename `var4` which is an inexcusable variable name